src/account.js

```diff
@@ -1,40 +1,61 @@
-import BigNumber from 'bignumber.js';
 import isString from 'lodash/isString';
+import BigNumber from 'bignumber.js';
+
 import { StrKey } from './strkey';
+import {
+  decodeAddressToMuxedAccount,
+  encodeMuxedAccountToAddress
+} from './util/decode_encode_muxed_account';
 
 /**
  * Create a new Account object.
  *
- * `Account` represents a single account in Stellar network and its sequence number.
- * Account tracks the sequence number as it is used by {@link TransactionBuilder}.
- * See [Accounts](https://stellar.org/developers/learn/concepts/accounts.html) for more information about how
- * accounts work in Stellar.
+ * `Account` represents a single account in the Stellar network and its sequence
+ * number. Account tracks the sequence number as it is used by {@link
+ * TransactionBuilder}. See
+ * [Accounts](https://stellar.org/developers/learn/concepts/accounts.html) for
+ * more information about how accounts work in Stellar.
+ *
  * @constructor
- * @param {string} accountId ID of the account (ex. `GB3KJPLFUYN5VL6R3GU3EGCGVCKFDSD7BEDX42HWG5BWFKB3KQGJJRMA`)
- * @param {string} sequence current sequence number of the account
+ *
+ * @param {string} accountId - ID of the account (ex.
+ *     `GB3KJPLFUYN5VL6R3GU3EGCGVCKFDSD7BEDX42HWG5BWFKB3KQGJJRMA` or
+ *     `MA7QYNF7SOWQ3GLR2BGMZEHXAVIRZA4KVWLTJJFC7MGXUA74P7UJUAAAAAAAAAAAAGZFQ`).
+ *     If you provide a muxed account address, only the underlying public key
+ *     (G... address) will be used.
+ * @param {string} sequence  - current sequence number of the account
  */
 export class Account {
   constructor(accountId, sequence) {
+    if (StrKey.isValidMed25519PublicKey(accountId)) {
+      accountId = encodeMuxedAccountToAddress(
+        decodeAddressToMuxedAccount(accountId, true),
+        false
+      );
+    }
+
```

Comment:
I guess I still feel like this should fail, instead of silently throwing away the account sub-id part of the muxed account. But not super strong either way, so... /shrug

Reply:
I think you're right; throwing makes more sense here. They should use a `MuxedAccount` for M-addresses.